frontend/src/main/components/Courses/CoursesForm.js

```diff
@@ -0,0 +1,156 @@
+    
+
+    return (
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialContents && (
+                <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="id">Id</Form.Label>
+                    <Form.Control
+                        data-testid={testIdPrefix + "-id"}
+                        id="id"
+                        type="text"
+                        {...register("id")}
+                        value={initialContents.id}
+                        disabled
+                    />
+                </Form.Group>
+            )}
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="name">Name</Form.Label>
+                <Form.Control
+                    data-testid={testIdPrefix + "-name"}
+                    id="name"
+                    type="text"
+                    isInvalid={Boolean(errors.name)}
+                    {...register("name", {
+                        required: "Name is required.",
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.name?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="school">School</Form.Label>
+                <Form.Control
+                    data-testid={testIdPrefix + "-school"}
+                    id="school"
+                    type="text"
+                    isInvalid={Boolean(errors.school)}
+                    {...register("school", {
+                        required: "School is required."
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.school?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="term">Term</Form.Label>
+                <Form.Control
+                    data-testid={testIdPrefix + "-term"}
+                    id="term"
+                    type="text"
+                    isInvalid={Boolean(errors.term)}
+                    {...register("term", {
+                        required: "Term is required."
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.term?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="start">Start (iso format)</Form.Label>
+                    <Form.Control
+                        data-testid={testIdPrefix + "-start"}
+                        id="start"
+                        type="datetime-local"
+                        isInvalid={Boolean(errors.start)}
+                        {...register("start", { required: true, pattern: isodate_regex })}
```

Comment:
See what happens if you remove this; my sense is that the form might still work the same.
```suggestion
                        {...register("start", { required: true })}
```